projects/pastanaga/src/lib/calendar/date-input.component.ts

```diff
@@ -0,0 +1,92 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, EventEmitter, Input, Output, ViewChild } from '@angular/core';
+import { markForCheck } from '../common/utils';
+import { TextfieldCommon } from '../textfield/textfield.common';
+import { format, isValid, isToday, isYesterday, getMonth } from 'date-fns';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+
+@Component({
+    selector: 'pa-date-input',
+    templateUrl: './date-input.component.html',
+    styleUrls: ['../popup/_popup.scss', './date-input.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DateInputComponent extends TextfieldCommon {
+    
+    @Input() dateHelp: string = '';
+    @Input() datePlaceholder: string = 'mm/dd/yyyy';
+    @Input() errorMessage: string = 'Invalid date (mm/dd/yyyy)';
+    @Input() inputId: string = '';
+    @Input() minDate?: Date;
```

Comment:
should be `min` to keep the same name as in calendar (consistency)

Reply:
As we talked before, here we cannot use `min` as the `TextfieldCommon` same name input has type `Number` instead of `Date`.